dms/models/dms_file.py

```diff
@@ -617,6 +655,30 @@ def _inverse_content(self):
             for vals, ids in updates.items():
                 self.browse(ids).write(dict(vals))
 
+    def _create_model_attachment(self, vals):
+        if "default_directory_id" in self._context:
+            default_directory_id = (
+                self.env["dms.directory"]
+                .sudo()
+                .browse(self._context["default_directory_id"])
+            )
+            vals["directory_id"] = default_directory_id.id
+        directory_id = self.env["dms.directory"].sudo().browse(vals["directory_id"])
+        if directory_id and directory_id.res_model and directory_id.res_id:
+            attachment_id = self.env["ir.attachment"].create(
+                {
+                    "name": vals["name"],
+                    "datas": vals["content"],
+                    "res_model": directory_id.res_model,
+                    "res_id": directory_id.res_id,
+                    "dms_file": True,
+                }
+            )
+            vals["attachment_id"] = attachment_id.id
```

Comment:
Instead of modifying current `vals` dictionary, it's always a better practice to copy the dictionary and return that modified copy.